backend/server.ts

```diff
@@ -277,7 +277,10 @@ io.on("connection", socket => {
 		playingUsers[socket.request.session.id] = gameId;
 		logger.info(`Info: User #${socket.request.session.id} successfully created game #${gameId}`);
 	});
-
+ socket.on("setUsername", (newUsername: string) => {
+	 userList[socket.request.session.id] = newUsername
+	 console.log("The username is " + newUsername);
+	 });
```

Comment:
Indentation is wrong, please use Prettier in VSCode so the whole code is consistent.